extra-cpp-bindings/src/wallectconnectregistry.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+use crate::error::GameSdkError;
+use crate::{ImageUrl, Platform, WalletEntry};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Registry {
+    listings: BTreeMap<String, Listing>,
+    count: i64,
+    total: i64,
+}
+
+impl Registry {
+    pub(crate) fn fetch_new() -> Result<Self, GameSdkError> {
```

Comment:
could this save and replace registry.json if found updates?

Reply:
where should it save the file?

Reply:
or should one provide an optional path?

Reply:
It is good to be configurable. For cpp project, maybe users want to put in into cache/temp folder. For unreal, it would be under project_dir.

Reply:
Yup, optional path is good, if None, put it to default path, otherwise, put it to the specified path.